crates/matrix-sdk-ui/src/room_list_service/state.rs

```diff
@@ -46,21 +49,49 @@ pub enum State {
     Terminated { from: Box<State> },
 }
 
-impl State {
+const DEFAULT_DELAY_BEFORE_RECOVER: Duration = Duration::from_secs(1800);
+
+/// The state machine used to transition between the [`State`]s.
+#[derive(Clone, Debug, PartialEq)]
+pub struct StateMachine {
```

Comment:
Can you move `State` inside `StateMachine` please?

Reply:
So that was my first try, and then I got lost in how to deal with observing the `State` and ownership trouble. I can try again but the change will probably be quite more invasive.

Reply:
Do you allow me to try please?

Reply:
Let me push my current state. I am not far, but I broke a sync indicator test so I think I messed up something regarding observable.

Reply:
Done !

Reply:
Oupsss it seems to break a lot of complement crypto tests.

Reply:
I think it was not the case before my (bad ^^) last refactor.